plugin/v2/plugin.go

```diff
@@ -149,13 +149,59 @@ next:
 				} else if !ok {
 					return fmt.Errorf("%q is not settable", s.prettyName())
 				}
-				// is it, so lets update the config in memory
+				// it is, so lets update the config in memory
 				updateConfigEnv(&p.PluginObj.Config.Env, &s)
 				continue next
 			}
 		}
 
-		//TODO: check devices, mount and args
+		// range over all the mounts in the manifest
+		for _, mount := range p.PluginObj.Manifest.Mounts {
+			// found the mount in the manifest
+			if mount.Name == s.name {
+				// is it settable ?
+				if ok, err := s.isSettable(allowedSettableFieldsMounts, mount.Settable); err != nil {
+					return err
+				} else if !ok {
+					return fmt.Errorf("%q is not settable", s.prettyName())
+				}
+
+				// it is, so lets update the config in memory
+				updateConfigMounts(p.PluginObj.Config.Mounts, &s)
+				continue next
+			}
+		}
+
+		// range over all the devices in the manifest
+		for _, device := range p.PluginObj.Manifest.Devices {
+			// found the device in the manifest
+			if device.Name == s.name {
+				// is it settable ?
+				if ok, err := s.isSettable(allowedSettableFieldsDevices, device.Settable); err != nil {
+					return err
+				} else if !ok {
+					return fmt.Errorf("%q is not settable", s.prettyName())
+				}
+
+				// it is, so lets update the config in memory
+				updateConfigDevices(p.PluginObj.Config.Devices, &s)
+				continue next
+			}
+		}
+
+		// found the name in the manifest
+		if p.PluginObj.Manifest.Args.Name == s.name {
```

Comment:
yes that what we discusses last time. this it's not just flags but complete args (order is important) it doesn't really make sense to update only the arg number 2.

You update all of them at once. They should be treated as a group.